### PR TITLE
feat: gql layer for tx_template

### PR DESCRIPTION
**Pull request by thevaibhav-dixit**

### CODE REVIEW

File: cala-ledger/src/tx_template/entity.rs [hunk 1]
```diff
@@ -367,20 +367,20 @@ mod tests {
...
             .build()
             .unwrap()];
-        let new_journal = NewTxTemplate::builder()
+        let new_tx_template = NewTxTemplate::builder()
             .id(TxTemplateId::new())
             .code("CODE")
             .tx_input(
                 NewTxInput::builder()
-                    .effective("date('2022-11-01')")
+                    .effective("DATE('2022-11-01')")
```
Comment (bodymindarts) [inline cala-ledger/src/tx_template/entity.rs:375]: Why was this changed?

File: cala-server/src/graphql/convert.rs [hunk 2]
```diff
@@ -82,6 +93,114 @@ impl From<cala_ledger::journal::JournalValues> for Journal {
...
+        Self {
+            name: value.name,
+            r#type: ParamDataType::from(value.r#type),
+            default,
+            description: value.description,
+        }
+    }
+}
+
+impl From<cala_ledger::tx_template::ParamDataType> for ParamDataType {
```
Comment (bodymindarts) [inline cala-server/src/graphql/convert.rs:174]: I think async-graphql has an option to generate this conversion when you declare the GQL run type

File: cala-server/src/graphql/schema.rs [hunk 3]
```diff
@@ -111,6 +112,105 @@ impl<E: MutationExtensionMarker> CoreMutation<E> {
...
+        let TxTemplateTxInput {
+            effective,
+            journal_id,
+            correlation_id,
+            external_id,
+            description,
+            metadata,
+        } = input.tx_input;
+        new_tx_input_builder
+            .effective(effective.as_ref().to_string())
```
Comment (bodymindarts) [inline cala-server/src/graphql/schema.rs:136]: Why this as_ref().to_string() repeating pattern. Can probably improve this by accepting a trait

File: cala-server/src/graphql/tx_template.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
+use async_graphql::*;
+
+use super::primitives::*;
+
+#[derive(SimpleObject)]
+pub(super) struct TxTemplate {
+    pub id: ID,
+    pub tx_template_id: UUID,
+    pub code: String,
+    pub params: Option<Vec<ParamDefinition>>,
```
Comment (bodymindarts) [inline cala-server/src/graphql/tx_template.rs:10]: Why option? It’s a Vec so it can just be empty

Comment (thevaibhav-dixit) [reply cala-server/src/graphql/tx_template.rs:10]: yep I was going to ask this question as to why we have this type in sqlx-ledger... I think we should change this to Vec<ParamDefinition>

File: cala-ledger/src/tx_template/entity.rs (hunk 1 above)
Comment (thevaibhav-dixit) [reply cala-ledger/src/tx_template/entity.rs]: was testing something. committed by mistake

**Review by bodymindarts — APPROVED** [review]
